tokio-trace/tokio-trace-core/src/metadata.rs

```diff
@@ -217,10 +245,42 @@ impl<'a> fmt::Debug for Metadata<'a> {
             .field("file", &self.file)
             .field("line", &self.line)
             .field("field_names", &self.fields)
+            .field("callsite_kind", &self.callsite_kind)
             .finish()
     }
 }
 
+#[repr(usize)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
```

Comment:
I don't think the `Copy` or `Hash` impls are strictly necessary here. 

Also, I think we may want to remove the `#[repr(usize)]` as this type doesn't _need_ to be at least a usize in length --- I'd rather allow rustc to use smaller representations if it chooses to.